accepted/00000-openapi.md

```diff
@@ -0,0 +1,184 @@
+**Start Date:** 13/12/2024
+
+### Summary
+Introduce the OpenAPI specification to describe and document Uyuni's HTTP API, replacing the existing static documentation. It aims to enhance user and developer experiences by providing interactive, standardized, and more accessible API documentation.
+
+### Motivation
+The adoption of OpenAPI aims to enhance both user and developer experiences by modernizing API documentation, making it more accessible, interactive, and aligned with industry standards. The current static API documentation lacks interactivity, making it difficult for users to explore and use API endpoints effectively. Integrating OpenAPI-generated interactive documentation, such as Swagger UI, directly into Uyuni’s Web UI allows users to explore, experiment with, and test API endpoints in real-time. This eliminates the need for additional programming effort and helps users achieve their goals more efficiently.
+
+OpenAPI is a widely recognized standard supported by a robust ecosystem of tools, including API validators, client generators, and testing frameworks. Transitioning to OpenAPI enables developers and users to leverage these tools, streamlining workflows and boosting productivity. Additionally, OpenAPI annotations integrate seamlessly with Java’s syntax and provide strong typing, reducing errors and simplifying maintenance compared to the current Javadoc-based approach, which relies on non-standard tags. This shift ensures consistent, standardized documentation while minimizing the effort required to maintain it.
+
+The expected outcome of this integration is to deliver Uyuni's API documentation as a valid OpenAPI JSON specification, complemented by an interactive Swagger UI integrated into Uyuni’s web interface, allowing users to explore and experiment with the API.
+
+
+### Detailed Design
+
+#### A. Describing the API
+Endpoints will be described using Swagger annotations, replacing the current Javadoc comments. Each API method will be annotated with Swagger elements such as `@Operation`, `@Parameter`, `@ApiResponse`, and `@Schema` to define essential details, including summaries, parameters, request bodies, and responses. For example:
+
+**Current documentation:**
+```
+    /*
+ 	* @apidoc.doc Update Content Environment with given label
+ 	* @apidoc.param #param_desc("string", "projectLabel", "Content Project label")
+ 	* @apidoc.param #param_desc("string", "envLabel", "Content Environment label")
+ 	* @apidoc.param
+ 	*  #struct_begin("props")
+ 	*  	#prop_desc("string", "name", "Content Environment name")
+ 	*  	#prop_desc("string", "description", "Content Environment description")
+ 	*  #struct_end()
+ 	* @apidoc.returntype $ContentEnvironmentSerializer
+ 	*/
+```
+
+**Proposed Swagger Documentation:**
+```
+    @Operation(
+        summary = "Update Content Environment",
+        tags = {"contentmanagement"},
+        description = "Update Content Environment with given label",
```

Comment:
During the migration process, we should try to generate the data to provide more information on how to use it on XML-RPC API also, so we can replace the existing static documentation and have more information to users.